kyo-core/shared/src/main/scala/kyo/chunks.scala

```diff
@@ -0,0 +1,464 @@
+            val indexed = this.toIndexed
+            Loops.indexed(Chunks.init[T]) { (idx, acc) =>
+                if idx < size then
+                    val v = indexed(idx)
+                    f(v).map {
+                        case true  => Loops.continue(acc.append(v))
+                        case false => Loops.done(acc)
+                    }
+                else
+                    Loops.done(acc)
+            }
+
+    final def dropWhile[S](f: T => Boolean < S): Chunk[T] < S =
+        if isEmpty then Chunks.init
+        else
+            val size    = this.size
+            val indexed = this.toIndexed
+            Loops.indexed(()) { (idx, _) =>
+                if idx < size then
+                    val v = indexed(idx)
+                    f(v).map {
+                        case true  => Loops.continue(())
+                        case false => Loops.done(indexed.dropLeft(idx))
+                    }
+                else
+                    Loops.done(Chunks.init)
+            }
+
+    final def changes: Chunk[T] =
+        changes(null)
+
+    final def changes(first: T | Null): Chunk[T] =
+        if isEmpty then Chunks.init
+        else
+            val size    = this.size
+            val indexed = this.toIndexed
+            @tailrec def loop(idx: Int, prev: T | Null, acc: Chunk[T]): Chunk[T] =
+                if idx < size then
+                    val v = indexed(idx)
+                    if v.equals(prev) then
+                        loop(idx + 1, prev, acc)
+                    else
+                        loop(idx + 1, v, acc.append(v))
+                    end if
+                else
+                    acc
+            loop(0, null, Chunks.init)
+    end changes
+
+    final def collect[U, S](pf: PartialFunction[T, U < S]): Chunk[U] < S =
+        if isEmpty then Chunks.init
+        else
+            val size    = this.size
+            val indexed = this.toIndexed
+            Loops.indexed(Chunks.init[U]) { (idx, acc) =>
+                if idx < size then
+                    val v = indexed(idx)
+                    if pf.isDefinedAt(v) then
+                        pf(v).map { u =>
+                            Loops.continue(acc.append(u))
+                        }
+                    else
+                        Loops.continue(acc)
+                    end if
+                else
+                    Loops.done(acc)
+            }
+
+    final def collectUnit[U, S](pf: PartialFunction[T, Unit < S]): Unit < S =
+        if isEmpty then ()
+        else
+            val size    = this.size
+            val indexed = this.toIndexed
+            Loops.indexed(()) { (idx, _) =>
+                if idx < size then
+                    val v = indexed(idx)
+                    if pf.isDefinedAt(v) then
+                        pf(v).andThen {
+                            Loops.continue(())
+                        }
+                    else
+                        Loops.continue(())
```

Comment:
I think we should add a `Loops.continueUnit` member to avoid allocations. Possibly `Loops.doneUnit` as well.

Reply:
good idea! I'll add it